configure.ml

```diff
@@ -870,6 +870,8 @@ let strip =
 let md5sum =
   select_command "Don’t know how to compute MD5 checksums…" [
     "md5sum", [], [ "--version" ];
+    (* Alpine Linux md5sum is really limited... *)
+    "md5sum", [], [ "Makefile" ];
```

Comment:
Why keep the `--version` line?

Reply:
That still seems the recommended way? I'm not happy depending on an actual file for this, but IMHO it is better than even trying to play shell tricks in the limited busybox env that Alpine provides. [That would be `echo foo | md5sum` or something like that]

Reply:
>I'm not happy depending on an actual file for this

Why?

Reply:
Because it depends on a file to detect a command.

Reply:
>Because it depends on a file to detect a command.

How's that a problem?

Reply:
> How's that a problem?

If the file moves the test fails.

Reply:
I'd be better if our primary test doesn't depend on the file. Anyways I've made my opinion of #6456 in https://github.com/ocaml/opam-repository/pull/11054 clear.

Reply:
We could also test md5sum without arguments if we redirect its stdin.

Reply:
Depending on a file *only on Alpine* is arguably worse than depending on it everywhere, because it means nobody will notice when this breaks.

Reply:
The task seems so simple: detect whether some md5sum like command exists on the system. The correct way to implement this is indeed piping some input to the candidate programs, detect pipe failure,  try one more time, and then report an error about the pipe failure and exit with the portable way to exit a process with failure.

You can use pipes in OCaml just fine without ever touching a "shell".

Why do I have to explain such basics?